pgx-examples/spi_srf/src/lib.rs

```diff
@@ -0,0 +1,145 @@
+
+
+"#,
+    name = "create_dog_daycare_example_table",
+);
+
+#[pg_extern]
+fn calculate_human_years() -> TableIterator<
+    'static,
+    (
+        name!(dog_name, String),
+        name!(dog_age, i32),
+        name!(dog_breed, String),
+        name!(human_age, i32),
+    ),
+> {
+    /*
+        This function is a simple example of using SPI to return a set of rows
+        from a query. This query will return the same rows as the table, but
+        with an additional column that is the dog's age in human years.
+    */
+    let query = "SELECT * FROM spi_srf.dog_daycare;";
+
+    let mut results = Vec::new();
+
+    Spi::connect(|client| {
+        let mut tup_table: pgx::SpiTupleTable = client.select(query, None, None);
+
+        while let Some(row) = tup_table.next() {
+            let dog_name: String = row["dog_name"].value().unwrap();
+            let dog_age: i32 = row["dog_age"].value().unwrap();
+            let dog_breed: String = row["dog_breed"].value().unwrap();
+            let human_age: i32 = dog_age * 7;
+            results.push((dog_name, dog_age, dog_breed, human_age));
+        }
+
+        Ok(Some(()))
+    });
+
+    TableIterator::new(results.into_iter())
+}
+
+#[pg_extern]
+fn filter_by_breed(
+    breed: &str,
+) -> TableIterator<
+    'static,
+    (
+        name!(dog_name, Option<String>),
+        name!(dog_age, Option<i32>),
+        name!(dog_breed, Option<String>),
+    ),
+> {
+    /*
+        This function is a simple example of using SPI to return a set of rows
+        from a query. This query will return the records for the given breed.
+    */
+
+    let query = "SELECT * FROM spi_srf.dog_daycare WHERE dog_breed = $1;";
+    let args = vec![(PgBuiltInOids::TEXTOID.oid(), breed.into_datum())];
+
+    let mut results = Vec::new();
+
+    Spi::connect(|client| {
+        let mut tup_table: pgx::SpiTupleTable = client.select(query, None, Some(args));
+
+        while let Some(row) = tup_table.next() {
+            results.push((
+                row["dog_name"].value(),
+                row["dog_age"].value(),
+                row["dog_breed"].value(),
+            ));
+        }
+
+        Ok(Some(()))
+    });
+
+    TableIterator::new(results.into_iter())
+}
+
+#[cfg(any(test, feature = "pg_test"))]
+#[pg_schema]
+mod tests {
+    use crate::calculate_human_years;
+    use pgx::*;
+
+    #[pg_test]
+    fn test_calculate_human_years() {
+        let mut results: Vec<(String, i32, String, i32)> = Vec::new();
+
+        results.push(("Fido".to_string(), 3, "Labrador".to_string(), 21));
+        results.push(("Spot".to_string(), 5, "Poodle".to_string(), 35));
+        results.push(("Rover".to_string(), 7, "Golden Retriever".to_string(), 49));
+        results.push(("Snoopy".to_string(), 9, "Beagle".to_string(), 63));
+        results.push(("Lassie".to_string(), 11, "Collie".to_string(), 77));
+        results.push(("Scooby".to_string(), 13, "Labrador".to_string(), 91));
```

Comment:
```suggestion
        results.push(("Scooby".to_string(), 13, "Great Dane".to_string(), 91));
```
![scooby doo](https://user-images.githubusercontent.com/46493976/191808800-a7ed09a0-5a82-48d2-8116-62cbf3204de9.png)

Reply:
Haha lol. I know Scooby is not a Labrador. But for the sake of the filtering example, I need multiple dogs of the same breed.